scripts/docker/content/set_env.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+# Copyright (c) Microsoft. All rights reserved.
+
+AUTH_SERVER_URL=""
+RESOURCE_TYPE=""
+AUTH_TOKEN=""
+
+# Acquires auth token for authroizating against key vault.
+_acquire_token() {
+    __set_keyvault_auth_server
+
+    local _temp=$(curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "client_id=$PCS_AAD_APPID&resource=https%3A%2F%2Fvault.azure.net&client_secret=$PCS_AAD_APPSECRET&grant_type=client_credentials" $AUTH_SERVER_URL/oauth2/token)
+    AUTH_TOKEN=$(__parse_json $_temp "access_token")
+}
+
+# Fetch key vault secret.
+_get_keyvault_secret() {
+
+    # Get a new token each time you access key vault.
+    _acquire_token
```

Comment:
Can't you acquire token and save it in memory? So that you don't need to do multiple times?

Reply:
For every request we are supposed to use new token. This is I think a security feature. That's how SDK works.